service/http.ts

```diff
@@ -8,20 +11,44 @@ export interface HttpError {
   };
 }
 
-async function http<T>(
+interface HttpConfig {
+  init?: Omit<RequestInit, "body">;
+  omitAuthentication?: boolean;
+}
+
+let _unauthenticatedErrorHandler: (
+  e: HttpError
+) => Promise<void> = async () => {};
+export const setUnauthenticatedErrorHandler = (
+  f: (e: HttpError) => Promise<void>
+) => {
+  _unauthenticatedErrorHandler = f;
+};
+
+function getAuthToken(): string {
+  const rawToken = window.localStorage.getItem("auth.token");
+  const token = rawToken && rawToken?.length > 0 ? JSON.parse(rawToken) : "";
+  return token;
+}
+
+async function http<TResponse>(
   endpoint: string,
   configOverride: RequestInit
-): Promise<T> {
+): Promise<TResponse> {
   const url = new URL(endpoint, API_URL).toString();
+  const httpAbortController = new AbortController();
   const config = {
     headers: {
       "content-type": "application/json",
     },
+    signal: httpAbortController.signal,
     ...configOverride,
   };
   const request = new Request(url, config);
   const response = await fetch(request);
 
+  setTimeout(() => httpAbortController.abort(), httpTimeout);
```

Comment:
Is this to emulate the old timeout in the grpc client? Could we not just leave this and the browser will timeout by default anyway?

Also, _if_ we are keeping this, it should probably be before the `fetch` above and have the timeout cleared, as we won't want the `abort` to be triggered if the fetch succeeded